tests/backup/backup_upgrade_test.go

```diff
@@ -204,3 +204,368 @@ var _ = Describe("{StorkUpgradeWithBackup}", func() {
 		CleanupCloudSettingsAndClusters(backupLocationMap, cloudAccountName, cloudCredUID, ctx)
 	})
 })
+
+var _ = Describe("{UpgradeScenario}", func() {
+	var (
+		contexts               []*scheduler.Context
+		appNamespaces          map[string][]string
+		backupLocationMap      map[string]string
+		cloudAccountUid        string
+		cloudAccountName       string
+		backupLocationName     string
+		backupLocationUid      string
+		srcClusterUid          string
+		dstClusterUid          string
+		preRuleNames           map[string]string
+		preRuleUids            map[string]string
+		postRuleNames          map[string]string
+		postRuleUids           map[string]string
+		numDeployments         int
+		backupWithoutRuleNames []string
+		backupWithRuleNames    []string
+		restoreNames           []string
+		schedulePolicyName     string
+		schedulePolicyUid      string
+		intervalInMins         int
+		singleNamespaces       []string
+		singleNSScheduleNames  []string
+		allNamespaces          []string
+		allNSScheduleName      string
+	)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("UpgradeScenario", "UpgradeScenario", nil, 0)
```

Comment:
I will add more precise description at the end of the review.

Reply:
Done.

testName: PXBackupEndToEndBackupAndRestoreWithUpgrade

testDescription: Validates end-to-end backup and restore operations with PX-Backup upgrade